src/main/scala/prci/ClockNodes.scala

```diff
@@ -8,7 +8,9 @@ import freechips.rocketchip.diplomacy._
 
 object ClockImp extends SimpleNodeImp[ClockSourceParameters, ClockSinkParameters, ClockEdgeParameters, ClockBundle]
 {
-  def edge(pd: ClockSourceParameters, pu: ClockSinkParameters, p: Parameters, sourceInfo: SourceInfo) = ClockEdgeParameters(pd, pu, p, sourceInfo)
+  def edge(pd: ClockSourceParameters, pu: ClockSinkParameters, p: Parameters, sourceInfo: SourceInfo) = {
+    ClockEdgeParameters("foo", pd, pu, p, sourceInfo)
```

Comment:
@hcook i don't know what to put for the name here. Is this never used? I did not end up with any "foo" signals in the design so hard to understand the code flow exactly here.

Reply:
You would have to add `name` as a parameter to the individual `ClockSinkParameters`; right now only groups get named, not individual sinks.

Alternatively, you/we could add the `nodePath` parameters that many of the other diplomatic protocol types have [e.g. interrupts](https://github.com/chipsalliance/rocket-chip/blob/bf0e85422db1ea1f1ef4160963c6d9858e254eb1/src/main/scala/interrupts/Parameters.scala#L28) and then use the node path to derive the name (from the val name of the node or the node's `parent` module). That seems like a useful extension but probably as a separate PR

Reply:
As far as usage goes, this code gets used to make edges that are presumed never to be a group, i.e. only ever can accept a single clock. You would see this for example in a sifive-blocks device that has an asynchronous clock crossing inserted between it and the bus. However, even in that case, a `ClockBundle` is produced, and it is only `ClockGroupBundles` that currently use the name information in creating `members`.

Reply:
It's not clear to me that you actually _had_ to add `name` to `ClockEdgeParameters` and `ClockBundleParameters`. You could have made `ClockGroupEdgeParameters.members` an e.g. `Map[String, ClockEdgeParameters]" and therbey kept the naming concept only at the "Group" level

Reply:
yep, have done that